src/main/scala/com/fulcrumgenomics/util/Metric.scala

```diff
@@ -205,6 +216,60 @@ object Metric extends LazyLogging {
   def writer[T <: Metric](writer: Writer)(implicit tt: ru.TypeTag[T]): MetricWriter[T] = new MetricWriter[T](writer)
 }
 
+
+class MetricSorter[Key <: Ordered[Key], T <: Metric](maxObjectsInRam: Int = MetricSorter.MaxInMemory,
+                                                     keyfunc: T => Key,
+                                                     tmpDir: DirPath = Io.tmpDir
+                                                    )(implicit tt: ru.TypeTag[T]) extends Sorter[T, Key](
+  maxObjectsInRam = maxObjectsInRam,
+  codec           = new MetricSorter.MetricSorterCodec[T](),
+  keyfunc         = keyfunc,
+  tmpDir          = tmpDir
+)
+
+object MetricSorter {
+  /** The default maximum # of records to keep and sort in memory. */
+  val MaxInMemory: Int = 1e6.toInt
+
+  class MetricSorterCodec[T <: Metric]()(implicit tt: ru.TypeTag[T])
+    extends Sorter.Codec[T] with LazyLogging {
+    private val clazz: Class[T]   = ReflectionUtil.typeTagToClass[T]
+    private val reflectiveBuilder = new ReflectiveBuilder(clazz)
+
+    private def fail(message: String,
```

Comment:
TODO: I want to find a nice way to not duplicate this `fail` method here and in the `Metric` object.